mathisfun/Complex.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+class Complex {
+public:
+
+	Complex();
+
+	Complex(double real, double imaginary);
+
+	double getReal() const;
+
+	void setReal(double real);
+
+	double getImaginary() const;
+
+	void setImaginary(double imaginary);
+
+	bool operator==(const Complex& other) const;
+
+	const Complex operator+(const Complex& other) const;
```

Comment:
Why does this method returns a const Complex?
If I want to add 2 complex nubmers and then add 5 to the return value - why shouldn't I do it? Is that bad usage?

Reply:
You would be able to. The const would only prevent you from assigning to the return value. Since the method is marked const, you can use it on a constant, for example the return value, so you could add to the return value